### PR TITLE
Add no-deployment flag to quickstart

### PR DESCRIPTION
**Pull request by ancaantochi**
Flag was added to be able to disable starting a deployment on the device after Edge was bootstrapped.
If flag --no-deployment is set it doesn't start modules deployment. Only agent module is started by default.
If flag --no-deployment is missing it starts a deployment for edge hub and temp sensor modules.

### CODE REVIEW

File: smoke/IotEdgeQuickstart/Program.cs [hunk 1]
```diff
@@ -48,6 +48,7 @@ be used.
                             Sockets, otherwise N/A
                             switch form uses local IP address as hostname
   --username                anonymous, or Key Vault if --registry is specified
+  --no-deployment           Start IoT Edge modules deployment (EdgeHub and custom modules)
```
Comment (avranju) [inline smoke/IotEdgeQuickstart/Program.cs]: Nit: Should the help text say something like: `Deploys only Edge Agent if set`.

Comment (darobs) [inline smoke/IotEdgeQuickstart/Program.cs]: I initially read this as "--no-deployment means Start IoT Edge modules deployment", but it makes sense when you realize starting a deployment is the default.  Would it make it more clear to reword to something like, "Flag not set, start IoT Edge modules deployment"?

Comment (damonbarry) [inline smoke/IotEdgeQuickstart/Program.cs]: The first word ("start") should **not** be capitalized, to be consistent with the rest.

Comment (damonbarry) [reply smoke/IotEdgeQuickstart/Program.cs]: ...and `EdgeHub` should have a space, i.e., `Edge Hub`.

Comment (damonbarry) [reply smoke/IotEdgeQuickstart/Program.cs]: I'm not too happy with how this whole help message reads, but for now this is the "Defaults" sections, so it's a list of behaviors when the various flags/parameters are _not_ set. So "flag not set" is implied.

Comment (damonbarry) [reply smoke/IotEdgeQuickstart/Program.cs]: This section is about the default behaviors when a given flag is _not_ set.

Comment (damonbarry) [reply smoke/IotEdgeQuickstart/Program.cs]: I'd vote for text like this: `Deploy Edge Hub and temperature sensor modules`

File: smoke/IotEdgeQuickstart/Program.cs [hunk 2]
```diff
@@ -92,6 +93,9 @@ class Program
         [Option("--leave-running=<All/Core/None>", CommandOptionType.SingleOrNoValue, Description = "Leave IoT Edge running when the app is finished")]
         public LeaveRunning LeaveRunning { get; } = LeaveRunning.None;
 
+        [Option("--no-deployment", CommandOptionType.NoValue, Description = "Doesn't start IoT Edge module deployment")]
```
Comment (damonbarry) [inline smoke/IotEdgeQuickstart/Program.cs]: For consistency with other descriptions, should be `Don't deploy Edge Hub and temperature sensor modules`.

File: smoke/IotEdgeQuickstart/Quickstart.cs [hunk 3]
```diff
@@ -47,13 +50,16 @@ public class Quickstart : Details.Details
                     await StartBootstrapper();
                     await VerifyEdgeAgentIsRunning();
                     await VerifyEdgeAgentIsConnectedToIotHub();
-                    await DeployTempSensorToEdgeDevice();
-                    await VerifyTempSensorIsRunning();
-                    await VerifyTempSensorIsSendingDataToIotHub();
-
-                    if (this.leaveRunning == LeaveRunning.Core)
+                    if (!noDeployment)
```
Comment (damonbarry) [inline smoke/IotEdgeQuickstart/Quickstart.cs]: Should be `this.noDeployment`, right?

File: smoke/IotEdgeQuickstart/Program.cs (hunk 1 above)
Comment (ancaantochi) [reply smoke/IotEdgeQuickstart/Program.cs]: Changed to "deploy Edge Hub and temperature sensor modules", as "not set" is implicit

---
In reply to: [203526314](https://github.com/Azure/iotedge/pull/37#discussion_r203526314) [](ancestors = 203526314)

**Pull request merged**